lib/cuda/cudaCopyFromRingbuffer.cpp

```diff
@@ -0,0 +1,74 @@
+#include "cudaCopyFromRingbuffer.hpp"
+
+#include "cudaUtils.hpp"
+#include "math.h"
+#include "mma.h"
+
+using kotekan::bufferContainer;
+using kotekan::Config;
+
+REGISTER_CUDA_COMMAND(cudaCopyFromRingbuffer);
+
+cudaCopyFromRingbuffer::cudaCopyFromRingbuffer(Config& config, const std::string& unique_name,
+                                               bufferContainer& host_buffers,
+                                               cudaDeviceInterface& device) :
+    cudaCommand(config, unique_name, host_buffers, device, "cudaCopyFromRingbuffer", "") {
+    _output_size = config.get<int>(unique_name, "output_size");
+    _ring_buffer_size = config.get<int>(unique_name, "ring_buffer_size");
+    _gpu_mem_input = config.get<std::string>(unique_name, "gpu_mem_input");
+    _gpu_mem_output = config.get<std::string>(unique_name, "gpu_mem_output");
+    signal_buffer = dynamic_cast<RingBuffer*>(
+        host_buffers.get_generic_buffer(config.get<std::string>(unique_name, "host_signal")));
+    signal_buffer->register_consumer(unique_name);
+
+    input_cursor = 0;
```

Comment:
I think this might be a problem when GPU buffer depth != 1, since there will be a few copies of this object.   Would it be possible to simply register the consumer/producer once in `inst==0`, and use the return values from the ring buffer for the cursor?

Reply:
Fixed -- needs cudaCommandState to hold the cursor (since I'm not yet tracking the cursors in the ringbuffer, and adding multi-consumer is going to change everything).